test/unit/org/apache/cassandra/index/sai/cql/VectorLocalRandPerfTest.java

```diff
@@ -0,0 +1,146 @@
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+import com.google.common.collect.ArrayListMultimap;
+import com.google.common.collect.Multimap;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import io.github.jbellis.jvector.vector.VectorSimilarityFunction;
+import org.apache.cassandra.cql3.UntypedResultSet;
+import org.apache.cassandra.db.ColumnFamilyStore;
+import org.apache.cassandra.db.Keyspace;
+import org.apache.cassandra.db.marshal.FloatType;
+import org.apache.cassandra.db.marshal.Int32Type;
+import org.apache.cassandra.db.marshal.VectorType;
+import org.apache.cassandra.dht.Murmur3Partitioner;
+import org.apache.cassandra.index.sai.SAITester;
+import org.apache.cassandra.index.sai.disk.v1.SegmentBuilder;
+import org.apache.cassandra.index.sai.utils.Glove;
+import org.assertj.core.data.Percentage;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class VectorLocalRandPerfTest extends VectorTester
+{
+    private static Glove.WordVector word2vec;
+    final int vectorCount = 5000;
+    final int limit = 200;
+    List<float[]> vectors;
+
+    @Override
+    public void beforeTest() throws Throwable
+    {
+        super.beforeTest();
+
+        createTable(String.format("CREATE TABLE %%s (pk int, str_val text, val vector<float, %d>, PRIMARY KEY(pk))", word2vec.dimension()));
+        createIndex("CREATE CUSTOM INDEX ON %s(val) USING 'StorageAttachedIndex'");
+        waitForIndexQueryable();
+
+
+        vectors = IntStream.range(0, vectorCount)
+                           .mapToObj(s -> randomVector())
+                           .collect(Collectors.toList());
+        int pk = 0;
+        for (float[] vector : vectors)
+            execute("INSERT INTO %s (pk, str_val, val) VALUES (?, 'A', " + vectorString(vector) + " )", pk++);
+        flush();
+    }
+
+    @BeforeClass
+    public static void loadModel() throws Throwable
+    {
+        word2vec = Glove.parse(VectorLocalRandPerfTest.class.getClassLoader().getResourceAsStream("glove.3K.50d.txt"));
+    }
+
+    @Test
+    public void randomizedPerfTest() throws Throwable
+    {
+        long start = System.nanoTime();
+        // query on-disk index
+        for (var q : vectors)
+        {
+            UntypedResultSet resultSet = search(q, limit);
+            assertDescendingScore(q, getVectorsFromResult(resultSet));
+        }
+        long end = System.nanoTime();
+        logger.info("Querying vectors took {} millisecond", (float)(end - start) / 1_000_000.0);
+
+    }
+
+
+    private UntypedResultSet search(float[] queryVector, int limit) throws Throwable
+    {
+        UntypedResultSet result = execute("SELECT * FROM %s ORDER BY val ann of " + Arrays.toString(queryVector) + " LIMIT " + limit);
+        assertThat(result.size()).isCloseTo(limit, Percentage.withPercentage(5));
+        return result;
+    }
+
+    private String vectorString(float[] vector)
```

Comment:
use the vector() helper method instead

Reply:
I'll remove this test. It is useful for profiling but not so much as a regular unit test